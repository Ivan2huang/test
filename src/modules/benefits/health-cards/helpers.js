/* eslint-disable import/prefer-default-export */
export const getQRCodeXML = ({
  fullName,
  membershipNumber,
  certificateNumber,
  insurerCode,
  insurerName,
  insurerType,
  policyNumber,
  expiryDate,
}) =>
  [
    '<?xml version="1.0"?>',
    '<medical_card>',
    '<Card_type>',
    insurerType,
    '</Card_type>',
    '<insurance_company_code>',
    insurerCode,
    '</insurance_company_code>',
    '<insurance_company_name>',
    insurerName,
    '</insurance_company_name>',
    '<insured_name>',
    fullName,
    '</insured_name>',
    '<policy_no>',
    policyNumber,
    '</policy_no>',
    '<certificate_no>',
    certificateNumber,
    '</certificate_no>',
    '<membership_no>',
    membershipNumber,
    '</membership_no>',
    '<expiry_date>',
    expiryDate,
    '</expiry_date>',
    '</medical_card>',
  ].join('');
