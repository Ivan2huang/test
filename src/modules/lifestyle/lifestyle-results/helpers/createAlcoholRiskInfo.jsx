import React from 'react';

import { AlcoholIcon, NoAlcoholIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createAlcoholRiskInfo = (intl, theme, alcoholRisk) => {
  const alcoholRiskMapping = {
    None: `${theme.lifeStyleResultOK}`,
    RegularDrinker: `${theme.lifeStyleResultWarning}`,
    HeavyDrinker: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'alcohol',
    color: alcoholRiskMapping[alcoholRisk],
    CardIcon:
      alcoholRisk === 'None' ? (
        <NoAlcoholIcon />
      ) : (
        <AlcoholIcon color={alcoholRiskMapping[alcoholRisk]} />
      ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.alcoholRiskCard.title',
      'Alcohol',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.subTitle.${alcoholRisk}`,
      alcoholRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.description.${alcoholRisk}`,
      alcoholRisk,
    ),
  };
};

export default createAlcoholRiskInfo;
