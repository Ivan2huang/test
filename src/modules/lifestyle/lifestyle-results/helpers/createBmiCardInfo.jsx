import React from 'react';

import Typography from '../../../../uiComponents/Typography';
import { formatMessage } from '../../../../helpers/helpers';

export const formatBmiScore = bmiScore => {
  const score = parseFloat(bmiScore);
  if (Number.isNaN(score)) {
    return '';
  }
  return (+bmiScore).toFixed(1);
};

const createBmiCardInfo = (intl, theme, bmiScore, bmiClass) => {
  const bmiColorMapping = {
    Healthy: `${theme.lifeStyleResultOK}`,
    Overweight: `${theme.lifeStyleResultWarning}`,
    Obese: `${theme.lifeStyleResultAlarm}`,
    Underweight: `${theme.lifeStyleResultWarning}`,
  };

  return {
    id: 'bmi',
    color: bmiColorMapping[bmiClass],
    CardIcon: (
      <Typography type="style-1" color={theme.grey3} fontWeight="bold">
        {formatBmiScore(bmiScore)}
      </Typography>
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.bmiCard.title',
      'BMI',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.subTitle.${bmiClass}`,
      bmiClass,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.description.${bmiClass}`,
      bmiClass,
    ),
  };
};

export default createBmiCardInfo;
