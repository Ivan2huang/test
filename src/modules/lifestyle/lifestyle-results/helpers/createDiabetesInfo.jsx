import React from 'react';

import { HeartIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createDiabetesInfo = (intl, theme, diabetesRisk) => {
  const alcoholRiskMapping = {
    Low: `${theme.lifeStyleResultOK}`,
    High: `${theme.lifeStyleResultWarning}`,
    VeryHigh: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'diabetes',
    color: alcoholRiskMapping[diabetesRisk],
    CardIcon: <HeartIcon color={alcoholRiskMapping[diabetesRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.diabetesCard.title',
      'Diabetes',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.subTitle.${diabetesRisk}`,
      diabetesRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.description.${diabetesRisk}`,
      diabetesRisk,
    ),
  };
};

export default createDiabetesInfo;
