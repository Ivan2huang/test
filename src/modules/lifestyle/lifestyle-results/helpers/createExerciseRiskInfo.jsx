import React from 'react';

import { ExerciseIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createExerciseRiskInfo = (intl, theme, exerciseRisk) => {
  const exerciseRiskMapping = {
    High: `${theme.lifeStyleResultOK}`,
    LowToModerate: `${theme.lifeStyleResultWarning}`,
    VeryLow: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'exercise',
    color: exerciseRiskMapping[exerciseRisk],
    CardIcon: <ExerciseIcon color={exerciseRiskMapping[exerciseRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.exerciseRiskCard.title',
      'Exercise',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.subTitle.${exerciseRisk}`,
      exerciseRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.description.${exerciseRisk}`,
      exerciseRisk,
    ),
  };
};

export default createExerciseRiskInfo;
