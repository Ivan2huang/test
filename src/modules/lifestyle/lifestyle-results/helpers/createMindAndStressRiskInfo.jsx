import React from 'react';

import { StressIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createMindAndStressRiskInfo = (intl, theme, mindAndStressRisk) => {
  const mindAndStressRiskMapping = {
    Normal: `${theme.lifeStyleResultOK}`,
    BeCareful: `${theme.lifeStyleResultWarning}`,
    AtRisk: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'mindAndStress',
    color: mindAndStressRiskMapping[mindAndStressRisk],
    CardIcon: (
      <StressIcon color={mindAndStressRiskMapping[mindAndStressRisk]} />
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.mindAndStressRiskCard.title',
      'Mind & stress',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.mindAndStressRiskCard.subTitle.${mindAndStressRisk}`,
      mindAndStressRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.mindAndStressRiskCard.description.${mindAndStressRisk}`,
      mindAndStressRisk,
    ),
  };
};

export default createMindAndStressRiskInfo;
