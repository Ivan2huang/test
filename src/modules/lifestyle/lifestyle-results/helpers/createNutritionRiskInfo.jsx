import React from 'react';

import { NutritionIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createNutritionRiskInfo = (intl, theme, nutritionRisk) => {
  const nutritionRiskMapping = {
    Low: `${theme.lifeStyleResultOK}`,
    Moderate: `${theme.lifeStyleResultWarning}`,
    High: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'nutrition',
    color: nutritionRiskMapping[nutritionRisk],
    CardIcon: <NutritionIcon color={nutritionRiskMapping[nutritionRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.nutritionRiskCard.title',
      'Nutrition',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.nutritionRiskCard.subTitle.${nutritionRisk}`,
      nutritionRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.nutritionRiskCard.description.${nutritionRisk}`,
      nutritionRisk,
    ),
  };
};

export default createNutritionRiskInfo;
