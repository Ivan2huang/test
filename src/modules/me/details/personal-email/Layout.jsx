import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import { Box, Card, withStyles } from '@material-ui/core';
import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos';

import Typography from '../../../../uiComponents/Typography';
import { formatMessage, navigateTo } from '../../../../helpers/helpers';
import paths from '../../../../helpers/paths';
import { DETAILS_ID } from '../../constant';

const styles = theme => ({
  backContainer: {
    display: 'flex',
    alignItems: 'center',
    cursor: 'pointer',
    marginTop: theme.spacing(5),
    marginBottom: theme.spacing(5),
    '& > button': {
      background: 'transparent',
      display: 'flex',
      border: 'none',
      alignItems: 'center',
    },
  },
  backIcon: {
    fontSize: theme.spacing(3),
  },
});

const Layout = ({ classes, intl, noBackground, children }) => {
  return (
    <>
      <Box className={classes.backContainer}>
        <button
          data-testid="btn-BackToDetails"
          type="button"
          onClick={() => navigateTo(paths.common[DETAILS_ID])}
        >
          <ArrowBackIosIcon className={classes.backIcon} />
          <Typography type="style-10">
            {formatMessage(intl, 'titles.me.details', 'My details')}
          </Typography>
        </button>
      </Box>
      {noBackground ? <Box>{children}</Box> : <Card>{children}</Card>}
    </>
  );
};

Layout.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  intl: PropTypes.shape({}).isRequired,
  children: PropTypes.node,
  noBackground: PropTypes.bool,
};

Layout.defaultProps = {
  children: '',
  noBackground: false,
};

export default withStyles(styles)(injectIntl(Layout));
