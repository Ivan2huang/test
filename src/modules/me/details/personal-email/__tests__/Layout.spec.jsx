import React from 'react';

import { render, fireEvent } from '@testing-library/react';
import withIntl from '../../../../../i18n/withIntlProvider';

import Layout from '../Layout';

import { navigateTo } from '../../../../../helpers/helpers';

jest.mock(
  '../../../../../uiComponents/Typography',
  // eslint-disable-next-line react/prop-types
  () => ({ children, ...rest }) => <div {...rest}>{children}</div>,
);

jest.mock('../../../../../helpers/helpers', () => ({
  navigateTo: jest.fn(),
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
}));

describe('Layout Component', () => {
  it('should match snapshot', () => {
    const Component = withIntl(Layout);
    const { container } = render(<Component>Form Detail</Component>);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with no background', () => {
    const Component = withIntl(Layout);
    const { container } = render(
      <Component noBackground>Form Detail</Component>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should navigate to details page', () => {
    const Component = withIntl(Layout);
    const { getByTestId } = render(<Component>Form Detail</Component>);

    fireEvent.click(getByTestId('btn-BackToDetails'));

    expect(navigateTo).toHaveBeenCalledWith('/me/details');
  });
});
