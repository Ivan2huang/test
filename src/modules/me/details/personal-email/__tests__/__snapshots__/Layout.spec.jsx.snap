// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-4 injectIntl(Layout)-backContainer-1"
  >
    <button
      data-testid="btn-BackToDetails"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root injectIntl(Layout)-backIcon-2"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
      <div
        type="style-10"
      >
        My details
      </div>
    </button>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    Form Detail
  </div>
</div>
`;

exports[`Layout Component should match snapshot with no background 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-45 injectIntl(Layout)-backContainer-42"
  >
    <button
      data-testid="btn-BackToDetails"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root injectIntl(Layout)-backIcon-43"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
      <div
        type="style-10"
      >
        My details
      </div>
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root-55"
  >
    Form Detail
  </div>
</div>
`;
